pkg/kubelet/kubeletconfig/checkpoint/download.go

```diff
@@ -46,16 +57,34 @@ type Payload interface {
 
 // RemoteConfigSource represents a remote config source object that can be downloaded as a Checkpoint
 type RemoteConfigSource interface {
-	// UID returns a globally unique identifier of the source described by the remote config source object
-	UID() string
 	// KubeletFilename returns the name of the Kubelet config file as it should appear in the keys of Payload.Files()
 	KubeletFilename() string
+
 	// APIPath returns the API path to the remote resource, e.g. its SelfLink
 	APIPath() string
-	// Download downloads the remote config source object returns a Payload backed by the object,
-	// or a sanitized failure reason and error if the download fails
-	Download(client clientset.Interface) (Payload, string, error)
-	// Encode returns a []byte representation of the NodeConfigSource behind the RemoteConfigSource
+
+	// UID returns a globally unique (space and time) identifier for the payload targeted by the source.
+	// This value is updated when Download is called.
+	UID() string
+
+	// ResourceVersion returns a resource version for the payload targeted by the source.
+	// This value is updated when Download is called.
```

Comment:
nit: Prefer describing what is returned, rather than behavior: e.g. `ResourceVersion returns the resource version of the most recently downloaded payload targeted by the source.`